addons/udes_stock/tests/test_refactoring_splitting.py

```diff
@@ -923,3 +923,75 @@ def test_split_mixed_products_partial_reserve(self):
         ]
         expected_move_format = [(self.apple, 2, 0), (self.banana, 3, 0)]
         self.assertCountEqual(move_format, expected_move_format)
+
+class TestRefactoringDateDone(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create stock: pallet with apples, pallet with bananas
+        create picking: for all of both
+        """
+        super().setUpClass()
+
+        # putaway is being refactored
+        cls.picking_type_putaway.write(
+            {
+                "u_post_assign_action": "group_by_move_line_key",
+                "u_move_line_key_format": "{package_id.name}",
+            }
+        )
+
+        cls.picking = cls.create_picking(cls.picking_type_in)
+
+        # Create two Goods In pickings
+        product_info_1 = [{"product": cls.apple, "qty": 5}]
+        product_info_2 = [{"product": cls.apple, "qty": 10}, {"product": cls.banana, "qty": 10}]
+
+        cls.pick_1 = cls.create_picking(cls.picking_type_in, products_info=product_info_1)
+        cls.pick_2 = cls.create_picking(cls.picking_type_in, products_info=product_info_2)
+
+    def test_does_not_propagate_date_done_to_incomplete_picking(self):
+        """Check the done date is not carried over to incomplete pickings"""
+        # Complete pick_1 and assert put away is generated and then assert that date done is set on the Goods In but not on the put away
+        self.pick_1.action_assign()
+        self.pick_1.move_line_ids.write(
+            {
+                "location_dest_id": self.received_location.id,
+                "qty_done": self.pick_1.move_line_ids.product_uom_qty,
+            }
+        )
+        self.pick_1.action_done()
+
+        self.assertEqual(self.pick_1.state, "done")
+        self.assertEqual(
+            self.pick_1.date_done, self.pick_1.move_lines.date
+        )  # do we care about the specific date done or just whether the value exists?
```

Comment:
A: We care about the specific date done.

Reply:
This fails in CI - probably because the pick date is not the same as the move lines date if it is done as normal. Once all the moves are done, then the date_done gets written (check i think action-done or something similar). So these are not necessarily equal - because the operations of the write occur at different points

Reply:
The time they are the same is in post_validate_refactor

Reply:
Will need to look a bit more into it, feels like we should do some sort of check above just seeing if anything is set.

Reply:
FTI: action_done in stock_picking:
        todo_moves._action_done()
        self.write({'date_done': fields.Datetime.now()}) 
so the times are different

Reply:
Maybe we could call action_done on the moves, check the picking info, then call action_done on the picking. Or do some mocking and check when the date_done write occurs

Reply:
What is wrong with just checking the date done exists?

Reply:
I think the point Boss Pete is making, is that due to refactoring, the write of the date_done on a picking happens lots. Which we don't want - in this case i think checking the date done before action-done is false then true and >= the last move date.

Reply:
We have `test02_maintain_single_pick_extra_info` in another test class which checks for date done, would that suffice do you think? (So long as it is passing)

Reply:
I think those tests split on the move and have grouping on post confirm so the refactor only happens on the split

Reply:
Have asked Caleb to look into the following test, with post validation refactoring only:

1. Do a picking for an Apple and a Banana and complete it
2. On post validation, refactor by product so we end up with 2 completed pickings instead of 1
3. Make sure both have the date done set (the new one should have a date done greater or equal to the original)